site/src/examples/drawer/OptionalCloseButton.tsx

```diff
@@ -0,0 +1,99 @@
+import { ChangeEvent, ReactElement, SyntheticEvent, useState } from "react";
+import { ComboBoxNext, Drawer, Option } from "@salt-ds/lab";
+import {
+  Button,
+  Checkbox,
+  FlexItem,
+  FormField,
+  FormFieldHelperText,
+  FormFieldLabel,
+  Input,
+  StackLayout,
+  H2,
+} from "@salt-ds/core";
+
+export const OptionalCloseButton = (): ReactElement => {
+  const [open, setOpen] = useState(false);
+  const [value, setValue] = useState("");
+
+  const postcodes = ["05011", "01050", "03040", "11050"];
+
+  const handleRequestOpen = () => {
+    setOpen(true);
+  };
+
+  const onOpenChange = (newOpen: boolean) => {
+    setOpen(newOpen);
+  };
+
+  const handleClose = () => {
+    setOpen(false);
+  };
+
+  const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
+    const value = event.target.value;
+    setValue(value);
+  };
+
+  const handleSelectionChange = (
+    event: SyntheticEvent,
+    newSelected: string[]
+  ) => {
+    return newSelected.length === 1 ? setValue(newSelected[0]) : setValue("");
```

Comment:
```suggestion
   setValue( newSelected.length === 1 ? newSelected[0] : "");
```
We don't need a return here and we can make it simpler by only using one `setValue`.